scripts/release/debian/deb_install.sh

```diff
@@ -42,12 +42,15 @@ setup() {
     set -v
     export DEBIAN_FRONTEND=noninteractive
     apt-get update
-    apt-get install -y apt-transport-https lsb-release gnupg curl
+    apt-get install --assume-yes --no-install-recommends apt-transport-https ca-certificates curl gnupg lsb-release
     set +v
 
     assert_consent "Add Microsoft as a trusted package signer?" ${global_consent}
     set -v
-    curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /etc/apt/trusted.gpg.d/microsoft.gpg
+    mkdir -p /etc/apt/keyrings
+    curl -sLS https://packages.microsoft.com/keys/microsoft.asc |
+      gpg --dearmor -o /etc/apt/keyrings/microsoft.gpg
```

Comment:
Using `-o` is causing issue if the file `/etc/apt/keyrings/microsoft.gpg` already exists: https://github.com/Azure/azure-cli/issues/28984

Reply:
yeah, it should have been ` gpg --yes --dearmor -o /etc/apt/keyrings/microsoft.gpg` to minimize the use of pipes and redirects, and it should have been caught when testing.